### DISCUSSION TITLE
Reduce dictionary entries shown?

### DISCUSSION THREAD

**Discussion by NicolasAyllon** [birchill/10ten-ja-reader#949] [Q&A] [answered]
I'm using the Firefox add-on and I was wondering if it's possible to reduce the number of dictionary entries shown to just 1? I'm a beginner student in Japanese, and seeing definitions for 3+ dictionaries feels a bit overwhelming. I also feel it makes the popup a bit too large for my screen.

I've tried unchecking several of the dictionaries in the add-on settings, but the popup still shows several entries.
<img width="452" alt="10ten - One Dictionary Selected" src="https://user-images.githubusercontent.com/51245273/165195159-b928d673-a80e-4007-8bfd-4717d9e4d969.png">
Is it possible to reduce the number of dictionary entries shown, or am I missing something? Thanks in advance!

**Comment by birtles** [comment, marked as answer]
Hi @NicolasAyllon!

Thanks for your message.

Let me check I understand your question:

Firstly, you can "hide" the name and kanji dictionary tabs by setting the tab position to "none":

![image](https://user-images.githubusercontent.com/1232595/165195841-ca9eeed0-c234-43d1-a37e-52350395e65e.png)

Secondly, you can also hide the different kanji references shown as per your screenshot.

However, I think what you are looking for is a way to limit the number of word entries shown. Is that correct?

If so, perhaps we could add settings like:

- [ ] Only show the longest matching words
- [ ] Hide uncommon words when there are multiple matches of the same length

I think it we added those settings, though, we might want to also add a shortcut for showing the extra matches?

**Reply by NicolasAyllon** [reply]
Thank you for your quick response! I think you are correct in that I am asking if it's possible to reduce the number of word entries shown (I was incorrect in assuming they were all from different dictionaries). They are sometimes are too long to fit on the screen, and I haven't yet figured how to get the popup to stick so I can scroll down (it disappears whenever I move the mouse).

I've made a rough sketch of what it currently looks like, and what reduced entries might look like:

**Current** (reality)
<img width="672" alt="10ten - many entries" src="https://user-images.githubusercontent.com/51245273/165204289-a002ddf0-f1bf-4560-a741-053ed917c325.png">

**Entries reduced** (sketch/edit)
<img width="672" alt="10ten - one entry" src="https://user-images.githubusercontent.com/51245273/165204298-f6ebf461-0155-4210-b67a-4aaa2aa8ec89.png">

I've unchecked all but 1 dictionary, so all these entries must be from that dictionary (but it still feels like too much information, for me, right now). Perhaps just showing the most common or longest word would be a helpful setting? Thank you for considering the feature.

**Reply by birtles** [reply]
> Thank you for your quick response! I think you are correct in that I am asking if it's possible to reduce the number of word entries shown (I was incorrect in assuming they were all from different dictionaries). They are sometimes are too long to fit on the screen, and I haven't yet figured how to get the popup to stick so I can scroll down (it disappears whenever I move the mouse).

This is something that needs work. By the look of the screenshots, it seems like 10ten thinks you're on a touch-screen device (hence the X in the top-right). If so, you can scroll by touching the screen.

However, longer term we want to make the popup interactive via mouse too -- possibly adding a "sticky mode" for that.

> I've made a rough sketch of what it currently looks like, and what reduced entries might look like:

Thanks for the sketches. That makes sense. It looks do-able but we'd just need to work out what approach to take.

We could introduce the two settings I mentioned, but it might be better to do something a bit more clever like adding a simple "Show only the most relevant results" setting, and letting the add-on filter the results based on its own heuristic. If it works well, we'd need to determine if it should be on by default or not.

> I've unchecked all but 1 dictionary, so all these entries must be from that dictionary (but it still feels like too much information, for me, right now).

Sorry, it's a bit confusing, but judging from your previous screenshot I think you have disabled all but one of the _kanji references_. That section just relates to the items shown up on the kanji tab. There are only three dictionaries: words, kanji, and names and it's not actually possible to disable them. I hope that makes sense!

**Reply by NicolasAyllon** [reply]
> However, longer term we want to make the popup interactive via mouse too -- possibly adding a "sticky mode" for that.

I'm using a laptop that also happens to have a touchscreen display, so maybe that is part of the cause. I'm not sure if this represents a lot of devices, but being able to scroll long popups would be helpful, thank you.

> We could introduce the two settings I mentioned, but it might be better to do something a bit more clever like adding a simple "Show only the most relevant results" setting, and letting the add-on filter the results based on its own heuristic. If it works well, we'd need to determine if it should be on by default or not.

That makes sense, I think that would be helpful for a future feature.

> Sorry, it's a bit confusing, but judging from your previous screenshot I think you have disabled all but one of the kanji references. That section just relates to the items shown up on the kanji tab. There are only three dictionaries: words, kanji, and names and it's not actually possible to disable them. I hope that makes sense!

Yes that makes sense now, thank you for explaining! I'll go ahead and mark this as answered.

**Reply by birtles** [reply]
> I'm using a laptop that also happens to have a touchscreen display, so maybe that is part of the cause. I'm not sure if this represents a lot of devices, but being able to scroll long popups would be helpful, thank you.

Yes, 10ten's not too helpful for that case. If it detects a touchscreen, it assumes you will use it to scroll and lays out the popup accordingly but it might not be the most convenient way to scroll. Perhaps it should look for a mouse too.

> That makes sense, I think that would be helpful for a future feature.

Great. I'll have to think about the particular rules to use to get the most useful selection of entries.

> Yes that makes sense now, thank you for explaining! I'll go ahead and mark this as answered.

Thank you!
